File: javascript/2016.js
/**
 * @param {number[]} n
 * @return {number}
 */
var maximumDifference = function (n) {
    let p = n[0], l = n.length, m = -1;
    for (let i = 1; i < l; i++)
        if (p < n[i])
            m = Math.max(m, n[i] - p);
        else
            p = n[i];
    return m;
};

File: cpp/2016.cpp
class Solution {
public:
    int maximumDifference(vector<int>& n) {
        int p = n[0], l = n.size(), m = -1;
        for (int i = 1; i < l; i++)
            if (p < n[i])
                m = max(m, n[i] - p);
            else
                p = n[i];
        return m;
    }
};
